lib/luadata.c

```diff
@@ -157,45 +167,69 @@ static int luadata_lnew(lua_State *L)
 
 	data->ptr = lunatik_checkalloc(L, size);
 	data->size = size;
-	data->free = true;
+	data->flags = LUADATA_FREE;
 	return 1; /* object */
 }
 
 LUNATIK_NEWLIB(data, luadata_lib, &luadata_class, NULL);
 
-lunatik_object_t *luadata_new(void *ptr, size_t size, bool sleep)
+lunatik_object_t *luadata_new(void *ptr, size_t size, bool sleep, uint32_t flags)
 {
 	lunatik_object_t *object = lunatik_createobject(&luadata_class, sizeof(luadata_t), sleep);
 
 	if (object != NULL) {
 		luadata_t *data = (luadata_t *)object->private;
 		data->ptr = ptr;
 		data->size = size;
-		data->free = false;
+		data->flags = flags;
 	}
 	return object;
 }
 EXPORT_SYMBOL(luadata_new);
 
-int luadata_reset(lunatik_object_t *object, void *ptr, size_t size)
+int luadata_reset(lunatik_object_t *object, void *ptr, size_t size, luadata_flags_t *flags)
 {
 	luadata_t *data;
 
 	lunatik_lock(object);
 	data = (luadata_t *)object->private;
 
-	if (data->free) {
+	if (data->flags & LUADATA_FREE) {
 		lunatik_unlock(object);
 		return -1;
 	}
 
 	data->ptr = ptr;
 	data->size = size;
+	if (flags != NULL)
+		data->flags = *flags;
+
 	lunatik_unlock(object);
 	return 0;
 }
 EXPORT_SYMBOL(luadata_reset);
 
+int luadata_getflags(lunatik_object_t *object, luadata_flags_t *flags)
+{
+	if (flags == NULL)
```

Comment:
Do we expect this function to be called with `flags=NULL`? If not, I suggest removing the validation and assuming the argument to be non-NULL by contract. Feel free to ignore this suggestion if you like.

Reply:
we moved this discussion to [matrix](https://matrix.to/#/!RHdYAjlWKfZRHOoOoB:matrix.org/$Oz5M8Aks4B1mffk6L881yyc90KYBCqRuRY0YC5eoln4?via=matrix.org&via=envs.net&via=nixserve.net).. please hop into there ;-)